pkg/cloudprovider/vsphereparavirtual/apis/nsxnetworking/v1alpha1/ippool_types.go

```diff
@@ -0,0 +1,82 @@
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// IPPool is the Schema for the ippools API.
+type IPPool struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	Spec   IPPoolSpec   `json:"spec"`
+	Status IPPoolStatus `json:"status,omitempty"`
+}
+
+// IPPoolSpec defines the desired state of IPPool.
+type IPPoolSpec struct {
+	// Subnets defines set of subnets need to be allocated.
+	Subnets []SubnetReq `json:"subnets"`
+}
+
+// IPPoolStatus defines the observed state of IPPool.
+type IPPoolStatus struct {
+	// Subnets defines subnets allocation result.
+	Subnets []SubnetRes `json:"subnets"`
+	// Conditions defines current state of the IPPool.
+	Conditions []IPPoolCondition `json:"conditions"`
+}
+
+// SubnetReq defines the subnet allocation request.
+type SubnetReq struct {
+	// PrefixLength defines prefix length for this subnet.
+	// +optional
+	PrefixLength int `json:"prefixLength,omitempty"`
+
+	// IPFamily defines the ip family type for this subnet, could be ipv4 or ipv6.
+	// +optional
+	IPFamily string `json:"ipFamily,omitempty"`
+
+	// Name defines the name of this subnet.
+	Name string `json:"name"`
+}
+
+// SubnetRes defines the subnet allocation result.
+type SubnetRes struct {
```

Comment:
The `Res` short-form is probably more confusing than helpful I think, can this be named just `SubnetResult`?